docs/sdk/src/reference_docs/frame_currency.rs

```diff
@@ -1,8 +1,55 @@
-//! FRAME Currency Abstractions and Traits
+//! # FRAME Currency Abstractions and Traits
 //!
-//! Notes:
+//! Currency related traits in FRAME provide standardized interfaces for implementing various
+//! currency functionalities. The fundamental reason to have these traits is to allow pallets to
+//! utilize a token without relying on the implementation detail. Or else we could have a perfectly
+//! fine runtime without these traits.
+//! These traits enable developers to create, transfer, and manage different forms of digital assets
+//! within the blockchain environment, ensuring that the economic aspects of the chain are robust
+//! and adaptable to different use cases.
+
 //!
-//! - History, `Currency` trait.
-//! - `Hold` and `Freeze` with diagram.
-//! - `HoldReason` and `FreezeReason`
-//! 	- This footgun: <https://github.com/paritytech/polkadot-sdk/pull/1900#discussion_r1363783609>
+//! ## The Currency Trait
+//!
+//! The [`Currency`][4] trait was initially introduced in Substrate to manage the native token
```

Comment:
> [`Currency`][4]

Nit but this is the first time I'm seeing this type of referencing in our docs, maybe we should just add the URL in line? I find it easier to just click on the word/expression directly if I want more details.

Reply:
Yes, it's a tradeoff... this format helps readability when there are long / too many links
https://github.com/paritytech/polkadot-sdk/pull/2683#discussion_r1444902876